app/models/post.rb

```diff
@@ -1,3 +1,14 @@
 class Post < ApplicationRecord
   belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+  after_save :update_post_counter
+
+  def update_post_counter
+    user.update(post_counter: user.posts.count)
```

Comment:
- [ ] The post belongs to an author, so we should use `author` here, not `user`.

- [ ] **[OPTIONAL]** Alternatively, you can user the `increment!` method for this implementation.

  ```
  author.increment!(:posts_counter)
  ```